### ISSUE TITLE
ConversationalRetrievalChain + Memory: Can only remember partial memory

### ISSUE THREAD

**Issue by hanqing-blend** [langchain-ai/langchain#4355]
### Issue you'd like to raise.

Hi, I'm creating a pdf chatbot with memory and the chain I use is ConversationalRetrievalChain + ConversationBufferMemory, but it seems that it can only remember some chat history. Does anyone have an idea of the reason?

It works well on a question like `can you summarize the answer?`,
but it cannot remember the first question I asked, 
```python
query = "Hi, what is my first question?"
result = qa({"question": query})
print(result['answer'])
```
```
I'm sorry, but I don't have access to that information as our conversation history is not stored. Is there anything else I can help you with?
```

Here is my code:
```python
def get_chat_history(inputs) -> str:
    res = []
    for human, ai in inputs:
        res.append(f"Human:{human}\nAI:{ai}")
    return "\n".join(res)

memory = ConversationBufferMemory(
        memory_key='chat_history', return_messages=True, output_key='answer')

# Create conversation chain that uses our vectordb as retriver, this also allows for chat history management
qa = ConversationalRetrievalChain.from_llm(ChatOpenAI(temperature=0,model_name='gpt-3.5-turbo'), db.as_retriever(),memory=memory,
    get_chat_history=get_chat_history,
    return_source_documents=True)

```

**Comment by heweapon** [comment]
I have same question! I understand as follow "The ConversationalRetrievalChain only uses message history to generate questions for the Retriever, but does not expose the history to the chat LLM. " So do you know how to solve it? Reply me if you know how to solve it. Thank you very much

**Comment by muhammadsr** [comment]
Sam issue here. No one to the rescue?
